libproxy/url.cpp

```diff
@@ -479,8 +479,7 @@ char* url::get_pac() {
 
 		// Get content
 		unsigned int recvd = 0;
-		buffer = new char[PAC_MAX_SIZE];
-		memset(buffer, 0, PAC_MAX_SIZE);
+		std::vector<char> dynamic_buffer;
```

Comment:
This keeps track of the size already, which makes `recvd` redundant and a source of bugs if they get out of sync. If  `dynamic_buffer` is truncated after short reads (which just adjusts the internal size) that could be avoided.